File: backend/app/schemas/inventory.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.core.enums import MovementType


class InventoryAdjustmentCreate(BaseModel):
    warehouse_id: uuid.UUID
    product_id: uuid.UUID
    quantity: int
    reason: str


class InventoryAdjustmentResponse(BaseModel):
    movement_id: uuid.UUID
    warehouse_id: uuid.UUID
    product_id: uuid.UUID
    movement_type: MovementType
    quantity: int
    balance_after: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class InventoryTransferCreate(BaseModel):
    from_warehouse_id: uuid.UUID
    to_warehouse_id: uuid.UUID
    product_id: uuid.UUID
    quantity: int


class InventoryTransferResponse(BaseModel):
    transfer_out_movement_id: uuid.UUID
    transfer_in_movement_id: uuid.UUID
    product_id: uuid.UUID
    quantity: int
    created_at: datetime


class InventoryResponse(BaseModel):
    warehouse_id: uuid.UUID
    product_id: uuid.UUID
    physical_stock: int
    reserved_stock: int
    damaged_stock: int
    expiry_stock: int
    sellable_stock: int

    model_config = ConfigDict(from_attributes=True)

File: frontend/types/inventory.ts
export interface InventoryResponse {
  warehouse_id: string;
  product_id: string;
  physical_stock: number;
  reserved_stock: number;
  damaged_stock: number;
  expiry_stock: number;
  sellable_stock: number;
}

export interface InventoryAdjustmentCreate {
  warehouse_id: string;
  product_id: string;
  quantity: number;
  reason: string;
}

export interface InventoryAdjustmentResponse {
  movement_id: string;
  warehouse_id: string;
  product_id: string;
  movement_type: string;
  quantity: number;
  balance_after: number;
  created_at: string;
}

export interface InventoryTransferCreate {
  from_warehouse_id: string;
  to_warehouse_id: string;
  product_id: string;
  quantity: number;
}

export interface InventoryTransferResponse {
  transfer_out_movement_id: string;
  transfer_in_movement_id: string;
  product_id: string;
  quantity: number;
  created_at: string;
}
